soh/CMakeLists.txt

```diff
@@ -1919,6 +1919,32 @@ if (CMAKE_SYSTEM_NAME STREQUAL "Windows")
 	set_target_properties(${PROJECT_NAME} PROPERTIES MSVC_RUNTIME_LIBRARY ${MSVC_RUNTIME_LIBRARY_STR})
 endif()
 ################################################################################
+# Find/download Boost
+################################################################################
+include(FetchContent)
+FetchContent_Declare(
+    Boost
+    URL      https://boostorg.jfrog.io/artifactory/main/release/1.81.0/source/boost_1_81_0.tar.gz
+    URL_HASH SHA256=205666dea9f6a7cfed87c7a6dfbeb52a2c1b9de55712c9c1a87735d7181452b6
+    SOURCE_SUBDIR "null" # Set to a nonexistent directory so boost is not built (we don't need to build it)
+    DOWNLOAD_EXTRACT_TIMESTAMP false # supress timestamp warning, not needed since the url wont change
+)
+
+set(Boost_NO_BOOST_CMAKE false)
+set(BOOST_INCLUDEDIR ${FETCHCONTENT_BASE_DIR}/boost-src) # Location where FetchContent stores the source
+message("Searching for Boost installation")
+find_package(Boost)
```

Comment:
looking at the docs here https://cmake.org/cmake/help/latest/module/FindBoost.html#examples it seems we can specify the components we're looking for from boost
```
find_package(Boost 1.66.0 COMPONENTS date_time filesystem system ...)
````

i also don't see where/if this is *linking* boost, it seems that's done via `ADDITIONAL_LIBRARY_DEPENDENCIES`, should we do that for the MT?

@dcvz you might have a better sense of what the answers to these questions should be

Reply:
From my understanding the `COMPONENTS` part is regarding libraries that need to be linked. In this case I do not need any Boost libraries, only the headers.

Reply:
ah, so everything from boost that's being pulled in is a pure header lib? that makes life easy!